### PR TITLE
Load submitted project on tab switch

### LINKED ISSUE #111: Load new on item submit
I'm working on a PR that covers a bulk of major updates - including spliting groups of projects into tabs/pages. Question: how should auto refresh work? Currently the one-page app fetches data every min and will load new projects into UI automatically. Do we wanna do similar things for the upcoming version as well? Does auto refresh keep user on the current tab/page or we want to switch them to the latest projects tab after auto refresh has happened?

/cc @xmatthewx @kristinashu

### PR DESCRIPTION
**Pull request by mmmavis**
Fixes #111

@cadecairos @acabunoc ✨ 

I removed a bunch of legacy code in this PR and I left inline comments for them.

Other than that, major things I did in this PR

- removed `PulseMaker.renderApp` method and moved relevant code to `PulseMaker.init` instead
- added callback to `PulseMaker.getData`
- added `PulseMaker._projectsUpdated` property
- closing "add new" form triggers data fetching
- on **tab switch**, check to see if there's updated project data available. Load updated data if exists.

To test,

1. try adding a new project
2. close form
3. navigate to the Latest Tab and see if your project appears

### CODE REVIEW

File: css/projects.css [hunk 1]
```diff
@@ -479,17 +479,6 @@ label, input[type='radio'] {
...
-
-#loading {  
-  width:100%;
-  height:200px; 
-  line-height:200px;
-  text-align: center;
-  font-size:4em;
-  color:rgba(0,0,0,0.67);
-}
-
```
Comment (mmmavis) [inline css/projects.css]: removed legacy code

File: js/pulse.js [hunk 2]
```diff
@@ -1,22 +1,5 @@
...
-/* better typography */
-
-var typography = {
-  preventTextOrphans : function () {
-    var textElements = document.querySelectorAll('h1,h3,li,p');
-    Array.prototype.forEach.call(textElements, function(el, i){
-      el.innerHTML = el.innerHTML.replace(/\s([^\s<]{0,10})\s*$/,'&nbsp;$1');
-    }); 
-  },
-};
```
Comment (mmmavis) [inline js/pulse.js]: removed legacy code

File: js/pulse.js [hunk 3]
```diff
@@ -128,16 +103,6 @@ var PulseMaker = {
...
-  'refresh' : function () {
-    console.log('refresh');
-    PulseMaker.getData(false);
-  },
-  'clearProjectLists' : function () {
-    document.getElementById('loading').style.display = 'block';
-  },
-  'toggleAdditionalFeatures' : function () {
-    if (FEATURE.orphans) { typography.preventTextOrphans(); }
-  },
```
Comment (mmmavis) [inline js/pulse.js]: `refresh`, `cleanProjectLists`, `toggleAdditionalFeatures` are all legacy code

File: js/views-manager.js [hunk 4]
```diff
@@ -198,6 +200,14 @@ var ViewsManager = {
 
     window.scrollTo(0,0);
   },
+  updateProjectData: function() {
+    var updatedProjects = PulseMaker.getProjectsUpdated();
+    if (updatedProjects) {
```
Comment (cadecairos) [inline js/views-manager.js]: nit:  return early here vs wrapping logic in an if

```
if (!thing) {
  return;
}
// do stuff
```

Comment (mmmavis) [inline js/views-manager.js]: awesome suggestion! thanks! let me update this PR now

**Comment by mmmavis** [comment]
@cadecairos any idea why though I already updated the code Github is still showing "changes requested"? 👀

**Comment by cadecairos** [comment]
I think it wante me to come back and "approve" the changes

**Comment by mmmavis** [comment]
@cadecairos yay thanks! and i should try out Github's new review features soon!

**Pull request merged**